src/linkml_map/transformer/object_transformer.py

```diff
@@ -234,11 +234,65 @@ def map_object(
                     aeval(slot_derivation.expr)
                     v = aeval.symtable["target"]
             elif slot_derivation.populated_from:
-                v = source_obj.get(slot_derivation.populated_from, None)
+                populated_from = slot_derivation.populated_from
```

Comment:
I like what you're doing here. Even better would be write a standalone object or method to implement this functionality (eventually with a view to centralizing in schemaview, but fine to keep local for now). It could be reused, and more easily tested with its own docstring